lib/dat-worker-pool/default_worker.rb

```diff
@@ -67,9 +61,11 @@ def work_loop
     #   callbacks. This ensures it causes the work loop to exit (see
     #   `work_loop`).
     def fetch_and_do_work
-      @on_sleep_callbacks.each{ |p| p.call(self) }
+      @runner.increment_worker_waiting
+      run_callback 'on_sleep'
       work_item = @queue.pop
-      @on_wakeup_callbacks.each{ |p| p.call(self) }
+      run_callback 'on_wakeup'
+      @runner.decrement_worker_waiting
```

Comment:
@kellyredding - Same with increment and decrement, now called instead of being a callback.